Massive code cleanup

Brycey92 opened the pull request:
I used IntelliJ IDEA's code analysis feature to help find warnings and bugs, and manually went through fixing them. The only warnings I let IDEA fix automatically were unused imports and most redundant casts.

ItemStacks have been converted to use ItemStack.EMPTY instead of null, to conform to new standards.

Annotations have been added to a LOT of the code to help devs determine when null can and can't be used, however they're nowhere near all-encompassing.

The inheritance of the BlockAdvancedBipropellantRocketMotor class was rearranged to prevent a potential mis-classification of them as monopropellant motors.

Vague fields and variables have been renamed where I found them, such as i -> itemStack or obj -> spaceObject. This excludes math variables that seemed intentionally named, and standardized names like loop control variables.

Lang files have been cleaned up to the best of my ability, but the non-en_US languages still have a lot of missing translation keys. Probably best for someone who speaks those languages to add the translations, rather than me with Google Translate.

There were some recipes that had invalid json, so I fixed them. This resulted in a few of the recipes changing slightly.

This commit should fix a lot of NPEs, though I didn't get them all.


**Next Steps:**
I highly recommend one of the original devs merge this and also run the source through their own IDE's code analysis, since there are still a lot of warnings I couldn't or didn't fix, and a large amount of them could be or turn into bugs.

The ItemStack changes affected some API methods. Mods developed against Advanced Rocketry should use builds containing this commit in the future, to prevent them from sending null ItemStacks. I did still split any changes to method names and prototypes into a separate branch that can be merged or ignored.

dmodoomsirius committed:
switch to dev builds of gc

dmodoomsirius committed:
Update README.md

dmodoomsirius committed:
add .github to 1.12 branch

voidsong-dragonfly committed:
Fix solar array rendering problems (#2033)

dmodoomsirius committed:
fix missing deps.gradle

dmodoomsirius committed:
fix compiling issues

voidsong-dragonfly committed:
Hopefully revert to FG2.3 in a working way

Brycey92 committed:
Fix gradle build (#2139)

* Fix gradle build

* Add output folder to .gitignore

Brycey92 committed:
Massive code cleanup

I used IntelliJ IDEA's code analysis feature to help find warnings and bugs, and manually went through fixing them. The only warnings I let IDEA fix automatically were unused imports and most redundant casts.
ItemStacks have been converted to use ItemStack.EMPTY instead of null, to conform to new standards.
Annotations have been added to a LOT of the code to help devs determine when null can and can't be used, however they're nowhere near all-encompassing.
The inheritance of the BlockAdvancedBipropellantRocketMotor class was rearranged to prevent a potential mis-classification of them as monopropellant motors.
Vague fields and variables have been renamed where I found them, such as i -> itemStack or obj -> spaceObject. This excludes math variables that seemed intentionally named, and standardized names like loop control variables.
Lang files have been cleaned up to the best of my ability, but the non-en_US languages still have a lot of missing translation keys. Probably best for someone who speaks those languages to add the translations, rather than me with Google Translate.
There were some recipes that had invalid json, so I fixed them. This resulted in a few of the recipes changing slightly.
This commit should fix a lot of NPEs, though I didn't get them all.

I highly recommend one of the original devs merge this and also run the source through their own IDE's code analysis, since there are still a lot of warnings I couldn't or didn't fix, and a large amount of them could be or turn into bugs.
The ItemStack changes affected some API methods. Mods developed against Advanced Rocketry should use builds containing this commit in the future, to prevent them from sending null ItemStacks. I did still split any changes to method names and prototypes into a separate branch that can be merged or ignored.
This commit requires a sister-commit for libVulpes, since the ItemStack refactoring affected methods there too.

voidsong-dragonfly committed:
Merge branch 'feature/nuclearthermalrockets' into 1.12-cleanup

Brycey92 commented:
This may introduce a bug where the Linker can link blocks to 2 or more rockets at the same time. No other functionality or edge cases of the Linker have been tested.

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/backwardCompat/WavefrontObject.java:
```diff
@@ -82,11 +82,11 @@ private void loadObjModel(InputStream inputStream) throws ModelFormatException
                 currentLine = currentLine.replaceAll("\\s+", " ").trim();
 
-                if (currentLine.startsWith("#") || currentLine.length() == 0)
+                /*if (currentLine.startsWith("#") || currentLine.length() == 0)
```
Why was this commented out?

Brycey92 replied:
The compiler deemed the `continue` redundant, since nothing else would be executed anyway if it was an empty `if` statement block or didn't exist.

voidsong-dragonfly replied:
Got it, cool. Dunno why that was there then

Brycey92 replied:
Probably more for the benefit of the original developer than for code flow.

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/block/plant/BlockLightwoodSapling.java:
```diff
@@ -41,36 +44,33 @@ public void generateTree(World world, BlockPos pos, IBlockState state,
+    @ParametersAreNonnullByDefault
     public void grow(World worldIn, Random rand, BlockPos pos, IBlockState state)
     {
-        this.grow(worldIn, pos, state, rand);
```
Why did this go away?

Brycey92 replied:
This function just rearranged the order of the parameters slightly before calling the original function, so I removed the unneeded function and un-rearranged the parameters in all calls.

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/inventory/GuiOreMappingSatellite.java:
```diff
@@ -55,10 +56,8 @@ public GuiOreMappingSatellite(SatelliteOreMapping satellite,EntityPlayer invento
 		playerPosZ = zCenter = (int) inventoryPlayer.posZ;
 
 		//Max zoom is 128
-		if(satellite != null) {
```
Why did this go away? The satellite can still be null if it's unlaunched I thought.

Brycey92 replied:
The compiler said the condition was always true. Following that thread, I found that the super class constructor `ContainerOreMappingSatellite` assumes that the satellite input is always nonnull. I could annotate that parameter to the constructor as `@Nullable` and put this `if` statement back if you believe the constructor can be called with null. Though we should really go through and plot out when to and not to use null for satellites, since it seems to have made it into savedata: https://github.com/Advanced-Rocketry/AdvancedRocketry/issues/2128

voidsong-dragonfly replied:
Hmm.... I swear I've had NPEs without that so... yeah.

Brycey92 replied:
It looks like the ContainerOreMappingSattelite would actually throw an NPE currently if it were passed a null SatteliteOreMapping, so a null would never make its way to the zoom calculations.

voidsong-dragonfly replied:
Ah....

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/item/components/ItemJetpack.java:
```diff
@@ -31,13 +31,14 @@
 
 public class ItemJetpack extends Item implements IArmorComponent, IJetPack {
 	
-	private static enum MODES {
```
Any reason why this became nonstatic?

Brycey92 replied:
The static modifier is redundant for inner enums.

voidsong-dragonfly replied:
Ah, forgot that

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/tile/TileRocketAssemblingMachine.java:
```diff
@@ -320,18 +319,20 @@ public void scanRocket(World world, BlockPos pos2, AxisAlignedBB bb) {
 									nuclearWorkingFluidUseMax += ((IRocketEngine) block).getFuelConsumptionRate(world, xCurr, yCurr, zCurr);
 									thrustNuclearNozzleLimit += ((IRocketEngine)block).getThrust(world, currBlockPos);
-								} else if (block instanceof BlockBipropellantRocketMotor || block instanceof BlockAdvancedBipropellantRocketMotor ) {
+								} else if (block instanceof BlockBipropellantRocketMotor /*|| block instanceof BlockAdvancedBipropellantRocketMotor*/) {
```
As long as this code works you don't need to leave the comments in

Brycey92 replied:
I'll remove the comments.

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/tile/TileUnmannedVehicleAssembler.java:
```diff
@@ -225,10 +221,10 @@ public void scanRocket(World world, BlockPos pos2, AxisAlignedBB bb) {
 									nuclearWorkingFluidUseMax += ((IRocketEngine) block).getFuelConsumptionRate(world, xCurr, yCurr, zCurr);
 									thrustNuclearNozzleLimit += ((IRocketEngine)block).getThrust(world, currPos);
-								} else if (block instanceof BlockBipropellantRocketMotor || block instanceof BlockAdvancedBipropellantRocketMotor ) {
+								} else if (block instanceof BlockBipropellantRocketMotor /*|| block instanceof BlockAdvancedBipropellantRocketMotor*/) {
```
Same as with standard assember

Brycey92 replied:
I'll remove the comments.

voidsong-dragonfly commented on src/main/java/zmaster587/advancedRocketry/dimension/DimensionProperties.java:
```diff
@@ -1760,12 +1734,12 @@ public void writeToNBT(NBTTagCompound nbt) {
 		nbt.setBoolean("isGasGiant", isGasGiant);
 		nbt.setBoolean("hasRings", hasRings);
-		nbt.setInteger("sealevel", sealevel);
+		nbt.setInteger("seaLevel", sealevel);
```
This was the only NBT name problem I found, I think I misread something else when I mentioned multiple before

Brycey92 replied:
I'll change it back

voidsong-dragonfly replied:
Already did it in my PR.

voidsong-dragonfly reviewed (changes requested):
See comments for needed fixes

voidsong-dragonfly committed:
Another batch of IDEA-prompted changes

Brycey92 committed:
Changes from "Massive code cleanup" code review

voidsong-dragonfly committed:
Multiple bugfixes

voidsong-dragonfly committed:
Fix warp core

voidsong-dragonfly committed:
Fix planet gen NPE

voidsong-dragonfly committed:
Fix data bus issue

voidsong-dragonfly committed:
More fixes!

 - Remove more problematic alt recipes
 - More cleanup
 - Remove useless/already resolved todos
 - Remove some more useless files
 - Remove multiple calls to deprecated methods and replace with correct one
 - Dedeprecate StorageChunk#getGuidanceComputer because there's no alternative and that works, plus the commit depping it doesn't make sense to do so

voidsong-dragonfly committed:
First batch of fixes

voidsong-dragonfly committed:
Second batch of fixes, with many IDEA mixups fixed

voidsong-dragonfly committed:
Merge branch '1.12-cleanup' into bryce_cleanup

Brycey92 committed:
Merge pull request #1 from Silfryi/bryce_cleanup

Another batch of IDEA-prompted changes

voidsong-dragonfly committed:
Merge branch 'feature/nuclearthermalrockets' into 1.12-cleanup

Brycey92 committed:
Attempt to fix FuelType NPE
